packages/demo/frontend/src/components/earn/EarnWithFrontendWallet.tsx

```diff
@@ -1,110 +1,397 @@
+          : undefined,
+        bundler: env.VITE_BASE_SEPOLIA_BUNDER_URL
+          ? {
+              type: 'simple',
+              url: env.VITE_BASE_SEPOLIA_BUNDER_URL,
+            }
+          : undefined,
+      },
+      {
+        chainId: optimismSepolia.id,
+      },
+    ],
+  } as unknown as ReactActionsConfig<T>
+}
+
+function useActionsMorpho<T extends ReactProviderTypes>(
+  hostedWalletProviderType: T,
+) {
+  const config = useMemo(
+    () => createActionsConfig(hostedWalletProviderType, 'morpho'),
+    [hostedWalletProviderType],
+  )
+  return useMemo(() => createActions(config), [config])
+}
+
+function useActionsAave<T extends ReactProviderTypes>(
+  hostedWalletProviderType: T,
+) {
+  const config = useMemo(
+    () => createActionsConfig(hostedWalletProviderType, 'aave'),
+    [hostedWalletProviderType],
+  )
+  return useMemo(() => createActions(config), [config])
+}
+
+const convertLendMarketToMarketInfo = (market: LendMarket): MarketInfo => {
+  const chainId = market.marketId.chainId
+
+  // Determine network info
+  let networkName = 'Unknown'
+  let networkLogo = '/base-logo.svg'
+  if (chainId === baseSepolia.id) {
+    networkName = 'Base Sepolia'
+    networkLogo = '/base-logo.svg'
+  } else if (chainId === optimismSepolia.id) {
+    networkName = 'Optimism Sepolia'
+    networkLogo = '/OP.svg'
+  }
+
+  // Determine provider logo
+  const providerLogo =
+    market.name.toLowerCase().includes('gauntlet') ||
+    market.name.toLowerCase().includes('morpho')
+      ? '/morpho-logo.svg'
+      : '/aave-logo-dark.svg'
+
+  // Determine asset info
+  const assetSymbol = market.asset.metadata.symbol
+  const assetLogo = assetSymbol.includes('USDC')
+    ? '/usd-coin-usdc-logo.svg'
+    : assetSymbol.includes('WETH')
+      ? '/eth.svg'
+      : '/usd-coin-usdc-logo.svg'
+
+  // Extract simple market name
+  const marketName = market.name.split(' ')[0] || market.name
+
+  return {
+    name: marketName,
+    logo: providerLogo,
+    networkName,
+    networkLogo,
+    assetSymbol,
+    assetLogo,
+    apy: market.apy.total,
+    isLoadingApy: false,
+    marketId: market.marketId,
+    provider: market.name.toLowerCase().includes('aave') ? 'aave' : 'morpho',
+  }
+}
+
 export function EarnWithFrontendWallet({
   wallet,
   selectedProvider,
   logout,
 }: EarnWithFrontendWalletProps) {
   const hostedWalletProviderType =
     FRONTEND_HOSTED_WALLET_PROVIDER_CONFIGS[selectedProvider]
-  const { actions } = useActions({ hostedWalletProviderType })
 
-  // Memoize operation functions to prevent infinite loops
-  const getTokenBalances = useCallback(
-    async () => wallet!.getBalance(),
-    [wallet],
+  // Primary Actions instance (Morpho) - used for balance operations
+  const morphoActions = useActionsMorpho(hostedWalletProviderType)
+
+  // Secondary Actions instance (Aave) - only used for Aave lend operations
+  const aaveActions = useActionsAave(hostedWalletProviderType)
```

Comment:
if working with two different lend providers requires two different hooks like this, I wonder if there are improvements we need to consider making to the sdk so that this is not required. this is not the best developer experience